bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/storeconfigexporter/StoreConfigExporterImpl.java

```diff
@@ -72,6 +83,10 @@ void initModel() {
         storeView = properties.get(STORE_CODE_PROPERTY, "default");
         graphqlEndpoint = properties.get(GRAPHQL_ENDPOINT_PROPERTY, "/magento/graphql");
 
+        // use the AEM publish server endpoint absolut URL only on author, if configured and if endpoint is not already an absolut URL
+        usePublishGraphqlEndpoint = properties.get(USE_PUBLISH_GRAPHQL_ENDPOINT_PROPERTY, false) && slingSettings.getRunModes().contains(
+            "author") && !StringUtils.startsWith(graphqlEndpoint, "http");
```

Comment:
There's an edge case. Customers could configure the GraphQL endpoint like this, which is a valid option and should not be translated by `Externalizer`: `//localhost:3002/graphql`.

Reply:
Updated the PR to cover this case as well.